src/main/java/com/puppycrawl/tools/checkstyle/checks/modifier/RedundantModifierCheck.java

```diff
@@ -225,12 +241,16 @@
             TokenTypes.CLASS_DEF,
             TokenTypes.ENUM_DEF,
             TokenTypes.RESOURCE,
+            TokenTypes.STRICTFP,
```

Comment:
Please take a look at content of this list/array.
All tokens are big structure blocks: method, class , variable, ..

And you put on the same level with such big structure blocks some modifier. There are a lot others modifiers like public, private, volatile,... .

We need new Check property like: targetJdkVersion.

If set to jdk17, consider TokenTypes.STRICTFP as redundant.

Reply:
@romani if list has big structures inside, that does not apply that there cannot be a modifier there. At least, it is tokens to inspect, not a list of class members.
I really do not see an issue with this. By checking higher types check will have additional complexity + false positives.

We discussed in original issue possibility to add some jdk version, this will require a lot more work + check should not be too smart to configure itself.

Reply:
please consider https://github.com/checkstyle/checkstyle/issues/10798#issuecomment-998368182